woodwork/logical_types.py

```diff
@@ -203,6 +207,34 @@ class Boolean(LogicalType):
 
     primary_dtype = "bool"
 
+    def __init__(self, cast_nulls_as=None):
+        if cast_nulls_as and not isinstance(cast_nulls_as, bool):
+            raise ValueError(
+                f"Parameter `cast_nulls_as` must be either True or False, recieved {cast_nulls_as}",
+            )
+        self.cast_nulls_as = cast_nulls_as
```

Comment:
I think it'd be helpful to store a dictionary for mapping the values back to the original values. That way we don't lose any information and it may help with data representation and serialization in tempo. Probably not necessary right now in this PR, but future work possibly.